src/components/common/NamedAddress/index.tsx

```diff
@@ -0,0 +1,21 @@
+import useAsync from '@/hooks/useAsync'
+import useChainId from '@/hooks/useChainId'
+import { getContract } from '@safe-global/safe-gateway-typescript-sdk'
+import EthHashInfo from '../EthHashInfo'
+import type { EthHashInfoProps } from '../EthHashInfo/SrcEthHashInfo'
+
+const NamedAddressInfo = ({ address, name, customAvatar, ...props }: EthHashInfoProps) => {
+  const chainId = useChainId()
+  const [contract] = useAsync(() => getContract(chainId, address), [address, chainId])
+
+  return (
+    <EthHashInfo
+      address={address}
+      name={name || contract?.displayName || contract?.name}
+      customAvatar={customAvatar || contract?.logoUri}
+      {...props}
+    />
+  )
+}
```

Comment:
Is this doing the same as the `onlyName` property we added to `EthHashInfo`? I couldn't test it since the swap widget is disabled on prod cgw and known names are not available on staging cgw

Reply:
It fetches the name from the CGW `/contract/{address}` endpoint if no name is given. 
I realized that we never fetch the contract name when we show the to address in "Interact with" fields of transactions or inside batch transactions.